dbms/src/Functions/FunctionBase64Conversion.h

```diff
@@ -0,0 +1,150 @@
+#include <Columns/ColumnConst.h>
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypeString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/GatherUtils/Algorithms.h>
+#include <IO/WriteHelpers.h>
+#include <Common/config.h>
+
+#if USE_BASE64
+#include <libbase64.h>
+
+
+namespace DB
+{
+using namespace GatherUtils;
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int INCORRECT_DATA;
+}
+
+struct Base64Encode
+{
+    static constexpr auto name = "base64Encode";
+    static constexpr auto buffer_size_multiplier = 5.0 / 3.0;
+};
+
+struct Base64Decode
+{
+    static constexpr auto name = "base64Decode";
+    static constexpr auto buffer_size_multiplier = 3.0 / 4.0;
```

Comment:
If all strings were empty, the result is also all empty strings and the result buffer have the same size, not 3/4.